### DISCUSSION TITLE
How to do file upload

### DISCUSSION THREAD

**Discussion by dmichiels** [joy-framework/joy#109] [Q&A]
Hello,
i'm discovering Joy, and I think I like it a lot!
For my first little project, I want to be able to do a file upload, but I don't really undersand how it works.
I do: 
-  [:input {:type "file" :name "thename"}]
- Add the middleware "file-uploads"

But then I'm lost.
Can someone help me?
Thank you in advance,
Denis

**Comment by swlkr** [comment]
I mean it's definitely possible multipart file uploads don't work, but on the off chance that they do you should have something like this in your handler's request :body

```clojure
@[{:filename "name of file" :content-type "content-type" :size 123 :tempfile "<file descriptor>"}]
```
